ably/realtime_channel.go

```diff
@@ -427,8 +427,8 @@ func (*subscriptionMessage) isEmitterData() {}
 //
 // See package-level documentation on Event Emitter for details about
 // messages dispatch.
-func (c *RealtimeChannel) Subscribe(ctx context.Context, name string, handle func(*Message)) (unsubscribe func(), err error) {
-	unsubscribe = c.messageEmitter.On(subscriptionName(name), func(message emitterData) {
+func (c *RealtimeChannel) Subscribe(ctx context.Context, name string, handle func(*Message)) (func(), error) {
```

Comment:
I am perfectly fine with using `non-named` returns for internal `functions`. That said, please feel free to close this thread if the meaning of return values is covered as a part of the doc.

Reply:
The whole point of named returns is so you can do an empty return right? This function never does that so it's kind of using named returns for no reason imo.

Reply:
> Since this is a user interfacing API, I am skeptical about how it will be shown to a user. I feel naming returns are more verbose while using a method.

Named returns are only in scope within the function itself, they aren't visible in the calling code, so the names themselves aren't really part of the user facing API.

If we think they provide documentation for how the return value can be interpreted, I think that information is better placed in the method comment.

Reply:
Thanks !!